### PR TITLE
circleci pipeline for build

### PR DESCRIPTION
**Pull request by forgondolin**

### CODE REVIEW

**Comment by pschoffer** [comment]
I would like to get this PR in first so that we have the build there already, but as it is it also contains `moving zip to mu-plugins` commit. Could you please clean that up, so that it is only the config change?

**Comment by forgondolin** [comment]
sure

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+  build:
+    docker:
+      - image: cibuilds/docker:19.03
+    steps:
+      - checkout
+      - setup_docker_php
+      - run:
+          name: "Build Docker Image"
+          command: docker build --file Dockerfile
+      - deploy:
```
Comment (pschoffer) [inline .circleci/config.yml]: The deploy step here could possibly conflict with docker hub integration. Please drop this for now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
\ No newline at end of file
+version: 2.0
+
+jobs:
+  build:
+    docker:
+      - image: cibuilds/docker:19.03
+    steps:
+      - checkout
+      - setup_docker_php
```
Comment (pschoffer) [inline .circleci/config.yml]: You have mentioned that this is not needed probably right?

**Review by pschoffer — CHANGES_REQUESTED** [review]
I left some comments. 

Also, somehow the job seems to not be triggered. May need to do a bit more of digging circleCI, could you please check. Maybe we need something like:
```
workflows:
  main:
    jobs:
      - build
```

File: .circleci/config.yml (hunk 2 above)
Comment (forgondolin) [reply .circleci/config.yml]: I totally forgot about it. Let me fix

Comment (forgondolin) [reply .circleci/config.yml]: I'm using this text as guide, I hope the version matches 
https://circleci.com/blog/build-cicd-piplines-using-docker/

**Comment by pschoffer** [comment]
Hi @forgondolin ,
I played with this a bit. I guess it is harder to debug this locally. You could use `circleci cli` if you would like. Anyway this is a sort solution that works. Dont want' to take away your PR so if you would just update it with that. I am happy to merge:)

```
version: 2.0

workflows:
  version: 2
  main:
    jobs:
      - build

jobs:
  build:
    docker:
      - image: docker:17.05.0-ce-git
    steps:
      - checkout
      - setup_remote_docker
      - run:
          name: "Build Docker Image"
          command: docker build  .
```

**Comment by forgondolin** [comment]
hmm, thats weird, im running that in a docker container... well, I'll fix it. sorry for the bad code

**Comment by pschoffer** [comment]
Perfect thanks mate!

**Pull request merged**